### PR TITLE
refactor(mqtt): on-demand ingress/egress configs

### PR DESCRIPTION
**Pull request by ysfscream**
image

image

### CODE REVIEW

File: src/types/rule.d.ts [hunk 1]
```diff
@@ -217,3 +213,5 @@ export interface RuleMetrics {
 }
 
 export type RuleDataItemWithMetrics = Merge<RuleItem, { metrics: Metrics }>
+
+export type BridgeItem = HTTPBridge & MQTTBridge
```
Comment (Kinplemelon) [inline src/types/rule.d.ts:217]: The `&` operator doesn't seem very accurate here, but it seems to reduce the number of errors reported?🤔

File: src/types/rule.d.ts [hunk 2]
```diff
@@ -155,10 +155,6 @@ export type MQTTIn = MQTTOut & {
   local_qos: QoSLevel
 }
 
-export type OtherBridge = Record<string, any>
-
-export type BridgeItem = HTTPBridge | MQTTOut | MQTTIn | OtherBridge
```
Comment (Kinplemelon) [inline src/types/rule.d.ts:160]: Why has otherBridge been removed here, is it meant to say that this type only exists on the enterprise branch?

File: src/types/rule.d.ts (hunk 1 above)
Comment (ysfscream) [reply src/types/rule.d.ts:217]: I don't feel like I can say it's inaccurate, it's more likely to be a dynamic type here, but if it's a union type, it has to be asserted every time, but it's not fixed every time here either, it's based on the value of the API bounce back, so I didn't think of a way to solve the type for now and used `&`

File: src/types/rule.d.ts (hunk 2 above)
Comment (ysfscream) [reply src/types/rule.d.ts:160]: I change back.

File: src/types/rule.d.ts (hunk 1 above)
Comment (ysfscream) [reply src/types/rule.d.ts]: I change back.

**Comment by ysfscream** [comment]
Something confused me

image

**Comment by Kinplemelon** [comment]
> Something confused me
> 
> image

I will check it

**Pull request merged**